### PR TITLE
Lightcurve Generator 

### PR DESCRIPTION
**Pull request by grazwegian**
Includes Science Data Lightcurve generator 
Fixes for compatibility with L1 FITS files

### CODE REVIEW

File: stix/idl/processing/spectrogram/stx_science_data_lightcurve.pro [hunk 1]
```diff
@@ -0,0 +1,225 @@
...
+    plot_obj = plotman(desc='STIX Lightcurve', input = ut_plot, /ylog, xst = 1)
+  endif
+
+  ;retrieve the data from the OSPEX object for the output structure
+  flux_str = ospex_obj->getdata(spex_units='flux')
+  ut2_time = ospex_obj->getaxis(/ut, /edges_2)
+  duration = ospex_obj->getaxis(/ut, /width)
+
+  light_curve_str = {$
+    data:flux_str.data, $
```
Comment (samaloney) [inline stix/idl/processing/spectrogram/stx_science_data_lightcurve.pro:134]: Maybe `counts` to indicate it not rate/livetime corrected. Also what the unit `counts/keV*cm^2` ?

**Review by samaloney — APPROVED** [review]

File: stix/idl/processing/spectrogram/stx_science_data_lightcurve.pro (hunk 1 above)
Comment (grazwegian) [reply stix/idl/processing/spectrogram/stx_science_data_lightcurve.pro:135]: Good point. I've updated to include the units. In the future this should be user selectable but right now the type is hard coded to flux which I think is the most generally meaningful.

**Review by samaloney — APPROVED** [review]

**Comment by afbattaglia** [comment]
Hi @grazwegian !
Thanks for the script, it is a great tool! If you want, I could create a pull-request in which I add some functionalities that I have had in my temporary script. But I guess this is not important for SPHERE so I could do it in two weeks or so. Next week we could discuss which functionality to add if you agree. What do you think?

**Comment by grazwegian** [comment]
Yes it would be really great if you could add some of your functionalities. We can discuss next week how it could fit together.

**Pull request merged**